mountinfo/mountinfo_filters.go

```diff
@@ -26,6 +29,11 @@ func PrefixFilter(prefix string) FilterFunc {
 // SingleEntryFilter looks for a specific entry
 func SingleEntryFilter(mp string) FilterFunc {
 	return func(m *Info) (bool, bool) {
+		// Resolve any symlinks in mountpoint, kernel would do the same and use the resolved path in /proc/<pid>/mountinfo.
+		if resolved, err := filepath.EvalSymlinks(mp); err == nil {
```

Comment:
This should probably be `filepath.Abs()`, which does it all.

And I think this is the responsibility of the caller to do that.

Reply:
I mixed it up, surely `filepath.Abs()` does not resolve symlinks, so you need `EvalSymlinks`, but maybe first you need to reject non-absolute paths.

And yes, I still think this is the responsibility of the caller.

Reply:
I agree that this is the caller’s responsibility。
Maybe we can execute `EvalSymlinks` in the `Mounted` function. Because Mounted returns whether a directory is a mountpoint.